homeassistant/util/yaml.py

```diff
@@ -185,12 +185,9 @@ def _ordered_dict(loader: SafeLineLoader,
 
         if key in seen:
             fname = getattr(loader.stream, 'name', '')
-            first_mark = yaml.Mark(fname, 0, seen[key], -1, None, None)
-            second_mark = yaml.Mark(fname, 0, line, -1, None, None)
-            raise yaml.MarkedYAMLError(
-                context="duplicate key: \"{}\"".format(key),
-                context_mark=first_mark, problem_mark=second_mark,
-            )
+            _LOGGER.warning(
```

Comment:
This is a util method which is not aware of Home Assistant. I agree that a persistent notification would be better but in that case we should see if we can add the warning to the returned object somehow. Let's do that in a future PR.

Following your suggestion now and changing it to be an error.